src/Streamly/Internal/Data/Fold/Types.hs

```diff
@@ -727,27 +730,31 @@ concatMap f (Fold stepa initiala extracta) = Fold stepc initialc extractc
             Done c -> return c
 
 ------------------------------------------------------------------------------
--- Internal APIs
+-- Mapping on input
 ------------------------------------------------------------------------------
 
--- | @(map f fold)@ maps the function @f@ on the input of the fold.
+-- | @(lmap f fold)@ maps the function @f@ on the input of the fold.
 --
--- >>> Stream.fold (Fold.map (\x -> x * x) Fold.sum) (Stream.enumerateFromTo 1 100)
+-- >>> Stream.fold (Fold.lmap (\x -> x * x) Fold.sum) (Stream.enumerateFromTo 1 100)
 -- 338350
 --
--- __Note__: This is not the same as 'fmap'. @map@ is contravariant where as
--- @fmap@ is covariant.
---
--- @since 0.7.0
-{-# INLINABLE map #-}
-map :: (a -> b) -> Fold m b r -> Fold m a r
-map f (Fold step begin done) = Fold step' begin done
+-- /Pre-release/
+{-# INLINABLE lmap #-}
+lmap :: (a -> b) -> Fold m b r -> Fold m a r
+lmap f (Fold step begin done) = Fold step' begin done
     where
     step' x a = step x (f a)
 
+-- XXX should be removed
```

Comment:
Why not remove it? I see that this PR has functional changes. You will remove (If exported then deprecate) it later?